api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -564,72 +560,59 @@ public RelationshipType saveRelationshipType(RelationshipType relationshipType)
 		if (viewType == null) {
 			return Context.getPersonService().getAllPersonAttributeTypes();
 		} else if (viewType == ATTR_VIEW_TYPE.LISTING) {
 			String patientListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_LISTING_ATTRIBUTES, "");
 			String userListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_LISTING_ATTRIBUTES, "");
-			if (personType == null || personType == PERSON_TYPE.PERSON) {
-				attrString = patientListing + "," + userListing;
-			} else if (personType == PERSON_TYPE.PATIENT) {
-				attrString = patientListing;
-			} else if (personType == PERSON_TYPE.USER) {
-				attrString = userListing;
-			} else {
-				log.error(MarkerFactory.getMarker("FATAL"), fatalString);
-			}
+      attrString = userPatientAttributes(userListing, patientListing, personType);
 		} else if (viewType == ATTR_VIEW_TYPE.VIEWING) {
 			String patientViewing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_VIEWING_ATTRIBUTES, "");
 			String userViewing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_VIEWING_ATTRIBUTES, "");
-			if (personType == null || personType == PERSON_TYPE.PERSON) {
-				attrString = patientViewing + "," + userViewing;
-			} else if (personType == PERSON_TYPE.PATIENT) {
-				attrString = patientViewing;
-			} else if (personType == PERSON_TYPE.USER) {
-				attrString = userViewing;
-			} else {
-				log.error(MarkerFactory.getMarker("FATAL"), fatalString);
-			}
+			attrString = userPatientAttributes(userViewing, patientViewing, personType);
 		} else if (viewType == ATTR_VIEW_TYPE.HEADER) {
 			String patientHeader = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_HEADER_ATTRIBUTES, "");
 			String userHeader = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_HEADER_ATTRIBUTES, "");
-			if (personType == null || personType == PERSON_TYPE.PERSON) {
-				attrString = patientHeader + "," + userHeader;
-			} else if (personType == PERSON_TYPE.PATIENT) {
-				attrString = patientHeader;
-			} else if (personType == PERSON_TYPE.USER) {
-				attrString = userHeader;
-			} else {
-				log.error(MarkerFactory.getMarker("FATAL"), fatalString);
-			}
-			
+			attrString = userPatientAttributes(userHeader, patientHeader, personType);
 		} else {
 			log.error(MarkerFactory.getMarker("FATAL"), fatalString);
 		}
 		
 		// the java list object to hold the values from the global properties
-		List<String> attrNames = new Vector<String>();
-		
-		// split the comma delimited string into a java list object
-		if (attrString != null) {
-			for (String s : attrString.split(",")) {
-				if (s != null) {
-					s = s.trim();
-					if (s.length() > 0) {
-						attrNames.add(s);
-					}
+		List<String> attrNames = new ArrayList<>();
+		if (StringUtils.isNotBlank(attrString)) {
+			for (String s : StringUtils.split(attrString, ",")) {
+				if (StringUtils.isNotBlank(s) && s.length() > 0) {
```

Comment:
I dont think `s.length > 0` is needed, if s is not blank it will be a non empty string.